### PR TITLE
Retrieve stripe card from source if it's not already a card

### LINKED ISSUE #6527: AttributeError: brand
https://sentry.prod.mozaws.net/operations/mdn-prod/issues/7302494/

```
Resolver404: {'tried': [[<RegexURLPattern None ^media/(?:redesign/)?css/(?P<doc>.*)-min.css$>], [<RegexURLPattern None ^media/(?:redesign/)?js/(?P<doc>.*)-min.js$>], [<RegexURLPattern None ^media/(?:redesign/)?img(?P<suffix>.*)$>], [<RegexURLPattern None ^media/(?:redesign/)?css(?P<suffix>.*)$>], [<RegexURLPattern None ^media/(?:redesign/)?js(?P<suffix>.*)$>], [<RegexURLPattern None ^media/(?:redesign/)?fonts(?P<suffix>.*)$>], [<RegexURLPattern None ^media/uploads/demos/(?:.*)$>], [<RegexURLPattern None (?i)^(?P<one>.*)//(?P<two>.*)//(?P<three>.*)$>], [<RegexURLPattern None (?i)^(?P<one>.*)//(?P<two>.*)$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/2_1_canvas_rect.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/2_2_canvas_moveto.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/2_3_canvas_lineto.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/2_4_canvas_arc.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/2_5_canvas_quadraticcurveto.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/2_6_canvas_beziercurveto.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/3_1_canvas_drawimage.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/3_2_canvas_drawimage.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/3_3_canvas_drawimage.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/3_4_canvas_gallery.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_1_canvas_fillstyle.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_2_canvas_strokestyle.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_3_canvas_globalalpha.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_4_canvas_rgba.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_5_canvas_linewidth.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_6_canvas_linecap.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_7_canvas_linejoin.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_8_canvas_miterlimit.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_9_canvas_lineargradient.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_10_canvas_radialgradient.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/4_11_canvas_createpattern.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/5_1_canvas_savestate.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/5_2_canvas_translate.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/5_3_canvas_rotate.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/5_4_canvas_scale.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/6_1_canvas_composite.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/6_2_canvas_clipping.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/globalCompositeOperation.html$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/backdrop.png$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/bg_gallery.png$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/gallery_1.jpg$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/gallery_2.jpg$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/gallery_3.jpg$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/gallery_4.jpg$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/gallery_5.jpg$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/gallery_6.jpg$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/gallery_7.jpg$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/gallery_8.jpg$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/picture_frame.png$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/rhino.jpg$>], [<RegexURLPattern None (?i)^samples/canvas-tutorial/images/wallpaper.png$>], [<RegexURLPattern None (?i)^samples/domref/mozGetAsFile.html$>], [<RegexURLPattern None (?i)^samples/raycaster/input.js$>], [<RegexURLPattern None (?i)^samples/raycaster/Level.js$>], [<RegexURL...
  File "redirect_urls/middleware.py", line 14, in __call__
    resolver_match = self.resolver.resolve(request.path_info)
  File "newrelic/hooks/framework_django.py", line 600, in wrapper
    return _wrapped(*args, **kwargs)
  File "newrelic/hooks/framework_django.py", line 588, in _wrapped
    result = wrapped(path)
  File "newrelic/hooks/framework_django.py", line 575, in wrapper
    return wrapped(*args, **kwargs)
  File "django/urls/resolvers.py", line 394, in resolve
    raise Resolver404({'tried': tried, 'path': new_path})

KeyError: 'brand'
  File "stripe/stripe_object.py", line 90, in __getattr__
    return self[k]
  File "stripe/stripe_object.py", line 131, in __getitem__
    raise err
  File "stripe/stripe_object.py", line 119, in __getitem__
    return super(StripeObject, self).__getitem__(k)

AttributeError: brand
(4 additional frame(s) were not displayed)
...
  File "django/views/decorators/cache.py", line 57, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "kuma/core/decorators.py", line 210, in wrapped
    return func(request, *args, **kwargs)
  File "kuma/users/views.py", line 472, in user_edit
    "subscription_info": retrieve_stripe_subscription_info(edit_user,),
  File "kuma/users/utils.py", line 53, in retrieve_stripe_subscription_info
    "brand": source.brand,
  File "stripe/stripe_object.py", line 92, in __getattr__
    raise AttributeError(*err.args)

AttributeError: brand
```

### PR DESCRIPTION
**Pull request by Gregoor**
Fixes #6527 

I didn't realize that the [`default_source`](https://stripe.com/docs/api/invoices/create#create_invoice-default_source) field was polymorphic and could also be of a Source type. It _could_ also have other types than the two handled here, but let's not get paranoid 🙊

### CODE REVIEW

File: kuma/users/utils.py [hunk 1]
```diff
@@ -46,14 +46,21 @@ def retrieve_stripe_subscription_info(user):
     )
     if stripe_subscription:
         source = stripe_customer.default_source
+        if source.object == "card":
+            card = source
+        elif source.object == "source":
+            card = source.card
+        else:
+            return None
```
Comment (peterbe) [inline kuma/users/utils.py]: Why this? 
I think I'd rather learn what that could be. E.g....
```suggestion
            raise NotImplementedError(f"don't know how to deal with {source.object!r}")
```

File: kuma/users/utils.py [hunk 2]
```diff
@@ -46,14 +46,21 @@ def retrieve_stripe_subscription_info(user):
     )
     if stripe_subscription:
         source = stripe_customer.default_source
```
Comment (peterbe) [inline kuma/users/utils.py]: What other sources are there? I.e. other than `default_source`

File: kuma/users/utils.py (hunk 1 above)
Comment (peterbe) [reply kuma/users/utils.py]: Hopefully, that'll either raise an error, which we learn from in Sentry, or it never happens.

File: kuma/users/utils.py (hunk 2 above)
Comment (Gregoor) [reply kuma/users/utils.py]: `stripe_customer` also has a list of sources one could iterate. But given that we control how sources are created, there are no other sources than Cards to be expected (BankAccounts would be another theoretical option)

File: kuma/users/utils.py (hunk 1 above)
Comment (Gregoor) [reply kuma/users/utils.py]: You're right, we should raise something here. I went with `ValueError`.

**Comment by Gregoor** [comment]
True, some test coverage would be great for this, but I don't believe in the stateless stripe stuff. What I'd like to look into is something like the headless tests we had for stripe before, but with a dependency-injection-like system for getting a test config in there. Imo the only downside of the headless test were noisier stripe interaction code and differing branches for test vs prod. That said, I don't know what our options are there yet. Could look into it more tomorrow.

File: kuma/users/utils.py [hunk 3]
```diff
@@ -51,7 +51,9 @@ def retrieve_stripe_subscription_info(user):
         elif source.object == "source":
             card = source.card
         else:
-            return None
+            raise ValueError(
+                f"unexpected stripe customer default_source of type {source.object}"
```
Comment (peterbe) [inline kuma/users/utils.py]:
```suggestion
                f"unexpected stripe customer default_source of type {source.object!r}"
```

Comment (peterbe) [reply kuma/users/utils.py]: Also, it says "type ..." but we're printing the *value*.  DOes that matter? Probably not.

Comment (Gregoor) [reply kuma/users/utils.py]: we're printing the value of `object` which specifies the type of `source`, so I'd argue it's correct

**Pull request merged**